### DISCUSSION TITLE
[Bug] Gemini-Pro 对话格式和 ChatGPT 一样，那 Gemini-Pro-Vision 呢？

### DISCUSSION THREAD

**Discussion by bbb3n** [lobehub/lobehub#1249] [General]
### 💻 系统环境

Windows

### 📦 部署环境

Vercel / Zeabur / Sealos

### 🌐 浏览器

Chrome

### 🐛 问题描述

我测试发现，我只填写了 OpenAI Key / Url （第三方中转）
gemini-pro 模型是可以按照 ChatGPT 格式调用的，这很棒
但是 gemini-pro-vision 无法按照 4v 的格式传图（会报错）

### 🚦 期望结果

希望如果不填写 Gemini 的 key 和 url ，仅填写 OpenAI 的 key 和 url 时：
**gemini-pro 和 gemini-pro-vision 均支持 OpenAI 的调用格式**

### 📷 复现步骤

问题描述已明确

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
>  gemini-pro 和 gemini-pro-vision 均支持 OpenAI 的调用格式

gemini-pro-vision 的调用格式和 openai 是不一样的，需要做转换：https://github.com/lobehub/lobe-chat/blob/main/src/libs/agent-runtime/google/index.ts#L67

你填上 Google API key 就能用了，为啥还要走 OpenAI
